
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://eaeomics.github.io/Hybrid-genome-assembly-practice/Laboratory_practice/">
      
      
      
        <link rel="next" href="Laboratory_practice/">
      
      
      <link rel="icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.44">
    
    
      
        <title>Hybrid genome assembly practice</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.0253249f.min.css">
      
        
        <link rel="stylesheet" href="assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
      
        <meta  property="og:type"  content="website" >
      
        <meta  property="og:title"  content="Hybrid genome assembly practice" >
      
        <meta  property="og:description"  content="None" >
      
        <meta  property="og:image"  content="https://eaeomics.github.io/Hybrid-genome-assembly-practice/Laboratory_practice/assets/images/social/index.png" >
      
        <meta  property="og:image:type"  content="image/png" >
      
        <meta  property="og:image:width"  content="1200" >
      
        <meta  property="og:image:height"  content="630" >
      
        <meta  property="og:url"  content="https://eaeomics.github.io/Hybrid-genome-assembly-practice/Laboratory_practice/" >
      
        <meta  name="twitter:card"  content="summary_large_image" >
      
        <meta  name="twitter:title"  content="Hybrid genome assembly practice" >
      
        <meta  name="twitter:description"  content="None" >
      
        <meta  name="twitter:image"  content="https://eaeomics.github.io/Hybrid-genome-assembly-practice/Laboratory_practice/assets/images/social/index.png" >
      
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="purple">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#hybrid-genome-assembly-practice" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="." title="Hybrid genome assembly practice" class="md-header__button md-logo" aria-label="Hybrid genome assembly practice" data-md-component="logo">
      
  <img src="assets/logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Hybrid genome assembly practice
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Hybrid-genome-assembly-practice
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="purple"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6m0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4M7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="lime"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    
  
  
    <li class="md-tabs__item md-tabs__item--active">
      <a href="." class="md-tabs__link">
        
  
    
  
  Hybrid-genome-assembly-practice

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="Laboratory_practice/" class="md-tabs__link">
        
  
    
  
  Rapid Sequencing Kit

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="." title="Hybrid genome assembly practice" class="md-nav__button md-logo" aria-label="Hybrid genome assembly practice" data-md-component="logo">
      
  <img src="assets/logo.png" alt="logo">

    </a>
    Hybrid genome assembly practice
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Hybrid-genome-assembly-practice
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="." class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Hybrid-genome-assembly-practice
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#context" class="md-nav__link">
    <span class="md-ellipsis">
      Context
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#intro" class="md-nav__link">
    <span class="md-ellipsis">
      Intro
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-1-read-inspection-and-qc" class="md-nav__link">
    <span class="md-ellipsis">
      Section 1: Read inspection and QC
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Section 1: Read inspection and QC">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#getting-the-data" class="md-nav__link">
    <span class="md-ellipsis">
      Getting the data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-inspection" class="md-nav__link">
    <span class="md-ellipsis">
      Read inspection
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-2-nanopore-draft-assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Section 2: Nanopore draft assembly
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Section 2: Nanopore draft assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#a-baseline-for-high-quality-assemblies" class="md-nav__link">
    <span class="md-ellipsis">
      A baseline for "high-quality" assemblies
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#draft-assembly-with-canu-nanopore-reads" class="md-nav__link">
    <span class="md-ellipsis">
      Draft assembly with Canu + Nanopore reads
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assembly-polishing-with-pilon" class="md-nav__link">
    <span class="md-ellipsis">
      Assembly polishing with Pilon
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-3-hybrid-assembly-unicycler" class="md-nav__link">
    <span class="md-ellipsis">
      Section 3: Hybrid assembly - Unicycler
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-4-nobody-expects-this-combo-unicycler-x-canu" class="md-nav__link">
    <span class="md-ellipsis">
      Section 4: Nobody expects this combo - Unicycler x Canu
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="Laboratory_practice/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Rapid Sequencing Kit
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="hybrid-genome-assembly-practice">Hybrid-genome-assembly-practice</h1>
<p>For full documentation visit <a href="http://bioinformatics-elearning.uab.cat:8080/moodle/">MSc Bioinformatics moodle</a>.</p>
<h2 id="context">Context</h2>
<p>Several patients have arrived at the hospital showing a common clinical presentation suggesting an infectious disease. Samples were collected and submitted to the laboratory and subjected to analyses for a complete panel of pathogens, which negative results for all the test. Given the urgency of the situation and the need of a clear diagnostic to implement a treatment to content the progress of infections, the laboratory isolated one strains and extracted the DNA to try to identify the pathogen through sequencing...</p>
<p>The lab contacts the head of the genomic department and request the sequencing of the strain to try to identify the species through the taxonomic assignation of the sequences. In addition to the species, information about virulence traits and antibiotic resistance have been also requested to be inferred from the “in silico” analyses of the strain genome.</p>
<p>To sequence the pathogen, a MinION device from Oxford Nanopore is available in your laboratory, although the team decided to obtain a second sequence using Illumina to contract with the Nanopore sequence and try to improve the quality of the genome. As your lab has not a Illumina sequencer, the DNA extraction is shipped to an external facility to be sequenced using Illumina requesting a “good quality” in terms of coverage (depth).
Now your boss has read that some crazy scientist have started to do something called... hybrid assembly. So she ask you to try this approach ...of course.</p>
<p><img src="assets/Designer.png" width="500"></p>
<h2 id="intro">Intro</h2>
<p>Assemble a genome!</p>
<p>Learn how to create and assess genome assemblies using the powerful combination of Nanopore and Illumina reads</p>
<p>This tutorial explores how long and short read data can be combined to produce a high-quality ‘finished’ bacterial genome sequence. Termed ‘hybrid assembly’, we will use read data produced from two different sequencing platforms, Illumina (short read) and Oxford Nanopore Technologies (long read), to reconstruct a bacterial genome sequence.</p>
<p>In this tutorial we will perform ‘de novo assembly’. De novo assembly is the process of assembling a genome from scratch using only the sequenced reads as input - no reference genome is used. This approach is common practise when working with microorganisms, and has seen increasing use for eukaryotes (including humans) in recent times.</p>
<p>Long reads can be used together with short reads to produce a high-quality assembly. Nanopore long reads (commonly &gt;40,000 bases) can fully span repeats, and reveal how all the genome fragments should be arranged. Therefore, while long reads will provide the general structure of the genome, short reads will provide that high base-level accuracy needed to close a genome.</p>
<p><strong>Data:</strong> Nanopore reads, Illlumina reads</p>
<p><strong>Tools:</strong> <code>Canu</code>, <code>Pilon</code>, <code>Unicycler</code>, <code>Quast</code>, <code>Busco</code>, <code>BWA</code>, <code>Samtools</code>,<code>FastQC</code>, <code>Trimmomatic</code>,<code>Cutadapt</code>,<code>Nanoplot</code>,<code>Filtlong</code>,<code>Porechop</code></p>
<h2 id="section-1-read-inspection-and-qc">Section 1: Read inspection and QC</h2>
<p>In this section we will import and perform quality control (QC) on our data. </p>
<p>Today we will use 4 pieces of data - <strong>2 short read sets, 1 long read set, and a reference genome</strong> (to compare our assembly with). </p>
<p><br></p>
<h3 id="getting-the-data">Getting the data</h3>
<p>For the illumina reads, the external organization has already uploaded the reads to the <strong>NCBI SRA database</strong> (Sequence Read Archive) which is the largest publicly available repository of high throughput sequencing data.</p>
<p>To download the data we will use the <code>fasterq-dump</code> tool from the SRA-toolkit which should be already installed in your conda environment.
We know that the accession ID of our samples is <strong>SRR10345480</strong> and <strong>SRR9042863</strong>. </p>
<p>Before we start typing and running commands, is important to generate a new directory where all the files are going to be saved and organized.
In our case we are going to create a directory called <code>hybrid_assembly</code> and a subdirectory called <code>data</code> in this subdirectory is where we are going to save our reads. <strong>Do not move them at any moment</strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>hybrid_assembly/data
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="nb">cd</span><span class="w"> </span>hybrid_assembly/data
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>fasterq-dump<span class="w"> </span><span class="o">{</span>accesion_ID<span class="o">}</span><span class="w"> </span><span class="c1">#repeat for both accesion IDs</span>
</span></code></pre></div>
<h3 id="read-inspection">Read inspection</h3>
<p>Often, it is prudent to first assess the quality of our read sets. For the short reads, we are concerned with base quality, sequence duplication, and presence of adapter sequences. For nanopore, we want to know about the length and quality distribution of reads, as these may both be highly variable. </p>
<p><code>FastQC</code> creates summary reports for short read data. If we were on a different situation with multiple samples and therefore, multiple fastqs, we could then use a tool called <code>MultiQC</code> to combine these reports for easy viewing. </p>
<p>For Nanopore data, <code>NanoPlot</code> is a great option. It creates plots which aim to summarise the length and quality distribution of long read sets. </p>
<p>Depending on these summaries, we may choose to perform a QC step to remove any poor quality reads before proceeding. </p>
<p><strong>Run FastQC</strong></p>
<p>As said before, we will use <code>FastQC</code> to see the quality of our <strong>Illumina</strong> reads. Once again, remember <strong>Organization is key</strong>, therefore we are going to create a new directory where we are going to save all our Quality Control outputs
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>qc/illumina_raw
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a><span class="nb">cd</span><span class="w"> </span>qc/illumina_raw
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>fastqc<span class="w"> </span>data/SRR*_1.fastq<span class="w"> </span>data/SRR*_2.fastq<span class="w"> </span>-o<span class="w"> </span>./
</span></code></pre></div></p>
<p>FastQC produces two outputs - 'RawData', and 'html'. Typically, the html is for human viewing, and the RawData can be given to other programs, such as MultiQC.</p>
<p>Let's see how the QC went on the Illumina Reads
<br>
<img src="assets/illuminaR1_raw.png">
<img src="assets/illuminaR2_raw.png">
<br></p>
<details>
<summary>Question 1(click to reveal)</summary>
What do you think about them? Do you think they have enough quality? Let's discuss, take your time to inspect the whole html
</details>

<p><strong>Clean Illumina reads</strong></p>
<p>It seems that our samples have some unwanted short reads and the quality of some bases is not as good as it could be.
There are plenty of programs such as <code>trimmomatic</code> or <code>cutadapt</code> that can be used to filter our Illumina reads.First we will get rid of the Illumina adapters, in this case we use Nextera adapters, using <code>Cutadapt</code>.
Then we will filter to the get the best possible reads by quality and length using <code>Trimmomatic</code> 
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>qc/trimmomatic
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>cutadapt<span class="w"> </span>-a<span class="w"> </span>CTGTCTCTTATACACATCT<span class="w"> </span>-A<span class="w"> </span>AGATGTGTATAAGAGACAG<span class="w"> </span>-o<span class="w"> </span>qc/trimmomatic/out_illumina_1.fastq<span class="w"> </span>-p<span class="w"> </span>qc/trimmomatic/out_illumina_2.fastq<span class="w"> </span>data/SRR9042863_1.fastq<span class="w"> </span>data/SRR9042863_2.fastq
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>trimmomatic<span class="w"> </span>PE<span class="w"> </span>-threads<span class="w"> </span><span class="m">4</span><span class="w"> </span>-phred33<span class="w"> </span>qc/trimmomatic/out_illumina_1.fastq<span class="w"> </span>qc/trimmomatic/out_illumina_2.fastq<span class="w"> </span>qc/trimmomatic/illumina_trimmed_1.fastq<span class="w"> </span>qc/trimmomatic/illumina_unpair_1.fastq<span class="w"> </span>qc/trimmomatic/illumina_trimmed_2.fastq<span class="w"> </span>qc/trimmomatic/illumina_unpair_2.fastq<span class="w"> </span>SLIDINGWINDOW:20:20<span class="w"> </span>MINLEN:250
</span></code></pre></div></p>
<p>We have generated two <strong>new fastqs</strong> called <code>illumina_trimmed_1.fastq</code> and <code>illumina_trimmed_1.fastq</code>. <img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.1.0/assets/svg/26a0.svg" title=":warning:" /> From now on when we say Illumina reads, these are the ones that we are talking about.
Now is time to see how this new fastqs differ from the others.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>qc/illumina_trimmed
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>fastqc<span class="w"> </span>qc/trimmomatic/illumina_trimmed_1.fastq<span class="w"> </span>qc/trimmomatic/illumina_trimmed_2.fastq<span class="w"> </span>-o<span class="w"> </span>qc/illumina_trimmed
</span></code></pre></div>
<p><br>
<img src="assets/illuminaR1_clean.png">
<img src="assets/illuminaR2_clean.png">
<br></p>
<p><img alt="‼" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.1.0/assets/svg/203c.svg" title=":bangbang:" /> Look at the difference in number of reads between the raw and the trimmed fastqs.</p>
<p><strong>Run NanoPlot</strong></p>
<p>As you know the reads of Nanopore are much longer than the Illumina ones, although in some cases (such as 16S experiments) we can use FastQC to look at them, it is better to use
a tool specifically designed for Nanopore, in this case we are going to use <code>Nanoplot</code></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>qc/nanoplot_raw
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="nb">cd</span><span class="w"> </span>qc/nanoplot_raw
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>NanoPlot<span class="w"> </span>-t<span class="w"> </span><span class="m">4</span><span class="w"> </span>--fastq<span class="w"> </span>../../data/SRR10*.fastq
</span></code></pre></div>
<p>Our median read length (7,365 pb) is quite good for Nanopore data,although some reads can be even longer, but the mean read quality is not that good (9.7), not so much time ago we would say that this quality is quite nice but the last generation of Nanopore flowcells produce much better reads, some of them at the same level as Illumina.
Now as we did before with the Illumina reads, let's clean the Nanopore reads.</p>
<p><strong>Clean Nanopore reads</strong></p>
<p>For cleaning our <code>Nanopore Reads</code> we will be using <code>Porechop</code>and <code>Filtlong</code> which are both of the most standardized software used for that. <code>Porechop</code>is the equivalent of
<code>Cutadapt</code> in Illumina and <code>Filtlong</code> is the equivalent of <code>Trimmomatic</code>.
Why do we need different programs to do that? Well, different technologies need different approaches, although some of the most popular software right now can support all <code>PacBio</code>, <code>Illumina</code>and <code>Nanopore</code>reads.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>qc/nanoplot_trimmed
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a><span class="nb">cd</span><span class="w"> </span>qc/nanoplot_trimmed
</span><span id="__span-5-5"><a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>
</span><span id="__span-5-6"><a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>porechop<span class="w"> </span>-i<span class="w"> </span>../../data/SRR10*.fastq<span class="w"> </span>-o<span class="w"> </span>./nanopore_adapter_clean.fastq
</span><span id="__span-5-7"><a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a>filtlong<span class="w"> </span>--min_length<span class="w"> </span><span class="m">1000</span><span class="w"> </span>--keep_percent<span class="w"> </span><span class="m">90</span><span class="w"> </span>--mean_q_weight<span class="w"> </span><span class="m">9</span><span class="w"> </span>nanopore_adapter_clean.fastq<span class="w"> </span>&gt;<span class="w"> </span>nanopore_filtered.fastq
</span></code></pre></div>
<p><img alt="‼" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.1.0/assets/svg/203c.svg" title=":bangbang:" /> As we did with the Illumina reads, from now on when we say nanopore reads we will be referring to <code>nanopore_filtered.fastq</code></p>
<p><strong>assign taxonomy of the pathogen</strong></p>
<p>Now that we have our reads clean, it is always interesting to know what are we looking for. Doing an assembly without knowing what are we facing could be terrible. For example,
imagine that you have assembled a genome with 6 closed chromosomes, first of all, congratulations because that is not easy, but then you assign somehow the taxonomy of that organism
and... WOW is a Ficus! 🍃 👏👏, but you rapidly realize that Ficus is 2n=26 and something has gone wrong.</p>
<p>There is plenty of ways to assign taxonomy when doing an assembly, for example one of the most used programs to assign taxonomy to illumina reads is <code>Kraken2</code>, another one used with Nanopore reads
could be <code>Emu</code>. But these programs require powerful machines, and let's be honest, your gaming laptop MSI i9 with 12 Cores and 32 GB of Ram is not as powerful as the clusters that are typically used for this
jobs (we are talking about &gt;40 Cores &gt;100GB ram).</p>
<p>To make things easy and rapid, we are going to do a trick. Taking advantage of the length of the nanopore reads, we will select the first read of the <code>fastq</code> file, or the second, it doesn't matter whilst is long enough, and they all should be after the QC that we have done, then we are going to blast it!</p>
<p><div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>head<span class="w"> </span>qc/nanopore_trimmed/nanopore_filtered.fastq<span class="w"> </span><span class="c1">#you can also open the file and select the first read</span>
</span></code></pre></div>
Now that you have copied the first read, go to <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&amp;PAGE_TYPE=BlastSearch&amp;LINK_LOC=blasthome">Blastn</a> and submit a job using the blastn option.
Wow is a <em>Vibrio parahaemolyticus</em>, thanks god is not that ficus again!
Go to the NCBI and download in <code>Fasta</code> format the <a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_009649015.1/">GCA_009649015.1</a> genome in a unique file.
Once you have download it, change the name of the fasta to <code>VP_reference_genome.fasta</code> and save it in the data directory.</p>
<h2 id="section-2-nanopore-draft-assembly">Section 2: Nanopore draft assembly</h2>
<h3 id="a-baseline-for-high-quality-assemblies">A baseline for "high-quality" assemblies</h3>
<p>To begin, we will identify what a high-quality assembly looks like.</p>
<p>When running assembly tools, we want to check the quality of assemblies we produce. It is paramount that genome assemblies are high-quality for them to be useful.</p>
<p>To get a baseline for what is considered a "high-quality" assembly, we will first run a common assembly QC tool - <code>Busco</code> - on a published genome similar to the organism we are working with today.</p>
<p><strong>Busco</strong></p>
<p><code>Busco</code> analysis uses the presence, absence, or fragmentation of key genes in an assembly to determine its quality.</p>
<p><code>Busco</code> genes are specifically selected for each taxonomic clade, and represent a group of genes that each organism in the clade is expected to possess. At higher clades, 'housekeeping genes' are the only members, while at more refined taxa such as order or family, lineage-specific genes can also be used.</p>
<p>We expect the reference genome to have all of these genes. When running <code>Busco</code>, we expect it to find most (if not all) of these in the assembly.</p>
<p>Let's create a new directory called <code>assembly_qc</code>. We want to create this directory outside from the <code>qc</code> directory we have been working on.</p>
<blockquote>
<p><strong>remember the directory structure we are using</strong></p>
<p>hybrid_assembly[data(fastqs), qc(illumina_raw, nanopore_raw, illumina_trimed, fastp), assembly_qc]</p>
</blockquote>
<p><div class="language-bash highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>assembly_qc
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a><span class="nb">cd</span><span class="w"> </span>assembly_qc
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>busco<span class="w"> </span>-i<span class="w"> </span>../../data/VP_reference_genome.fasta<span class="w"> </span>-l<span class="w"> </span>vibrionales<span class="w"> </span>-o<span class="w"> </span>busco_reference<span class="w"> </span>--augustus<span class="w"> </span>--mode<span class="w"> </span>genome<span class="w"> </span>
</span></code></pre></div>
After the program has run, look at the ‘short summary’ output. It may look something like this:</p>
<p><img src="assets/Busco_reference.png"></p>
<p>It seems that BUSCO could find almost all expected genes in the reference genome assembly.
By looking at the results, we see that we have 1443 / 1444 Complete BUSCOs, one being complete and duplicates, and another one missing.</p>
<p>This will form the baseline for the BUSCO QC results expected of a high-quality genome assembly.</p>
<p>From here, we will use our input DNA sequence data to assemble the genome of the sequenced organism, and will compare the QC results to that of the published <code>VP_reference_genome.fasta</code> assembly.</p>
<h3 id="draft-assembly-with-canu-nanopore-reads">Draft assembly with Canu + Nanopore reads</h3>
<p>Our first assembly will use the long-read data to create a draft genome, then the short-read data to "polish" (improve) the draft into a better assembly.</p>
<p>We will start by using a long-read assembly tool called <code>Canu</code> to create an assembly using the Nanopore long-read data.</p>
<p>Once again we need to create an specific directory for that, let's move again to the general <code>hibrid_assembly</code> directory
we will create a directory called <code>assemblies</code> where we will compute all the assemblies in this practice</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>assemblies/canu
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>canu<span class="w"> </span>-p<span class="w"> </span>canu<span class="w"> </span>-d<span class="w"> </span>assemblies/canu<span class="w"> </span>-nanopore<span class="w"> </span>qc/nanopore_trimmed/nanopore_filtered.fastq<span class="w"> </span>--genomeSize<span class="o">=</span><span class="m">5</span>.1m
</span></code></pre></div>
<p>We need to compare this assembly with our reference genome. For that we are going to use <code>Busco</code> again and `Quast</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>
</span><span id="__span-9-3"><a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>assembly_qc/canu_nanopore
</span><span id="__span-9-4"><a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a><span class="nb">cd</span><span class="w"> </span>assembly_qc/canu_nanopore
</span><span id="__span-9-5"><a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a>
</span><span id="__span-9-6"><a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a>busco<span class="w"> </span>-i<span class="w"> </span>../../assemblies/canu/canu.contigs.fasta<span class="w"> </span>-l<span class="w"> </span>vibrionales<span class="w"> </span>-o<span class="w"> </span>busco<span class="w"> </span>--augustus<span class="w"> </span>--mode<span class="w"> </span>genome<span class="w"> </span>--cpu<span class="w"> </span><span class="m">4</span>
</span></code></pre></div>
<p><img src="assets/Busco_draft_nanopore.png"></p>
<details>
<summary>Question 2(click to reveal)</summary>
As you can see we do not get the same output as the reference, Why do you think this is happening?
</details>

<p><strong>Quast</strong></p>
<p>Aside from <code>Busco</code>, we can use another method to perform assembly QC. In this case <code>Quast</code>allows us to compare two assemblies to determine their similarity.</p>
<p>We know that Bacteria can differ a lot between individuals of the same species, nevertheless they should have the same number of chromosomes with similar lengths. What we are going to do know is compare both <code>VP_reference_genome.fasta</code>and this new draft assembly based on Nanopore.</p>
<p>You know what is coming next, Yes! another directory <img alt="🥳" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.1.0/assets/svg/1f973.svg" title=":partying_face:" /></p>
<p><div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1">#in the same assembly_qc/canu_nanopore directory as before</span>
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>
</span><span id="__span-10-3"><a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>quast<span class="w"> </span>../../assemblies/canu/canu.contigs.fasta<span class="w"> </span>-r<span class="w"> </span>../../data/VP_reference_genome.fasta<span class="w"> </span>-o<span class="w"> </span>quast
</span></code></pre></div>
The output of <code>Quast</code> should be a directory with a set of files, we are interested in the <code>report.html</code> file
Now you should be looking at something like this:</p>
<p><img src="assets/quast_draft_nanopore.png"></p>
<p>what are we looking for? Well there is plenty of information here but you should take a closer look to:
* Genome fraction
* Number of contigs
* Number of misassemblies
* Number of mismatches per 100Kbp
* Number of indels per 100Kbp
* Total length
* Largest contig
* L50
* N50
* LGA50
* NGA50</p>
<details>
<summary>Question 3(click to reveal)</summary>
Which is the differnce between L50 and LGA50? What about N50 and NGA50?
</details>

<details>
<summary>Question 4(click to reveal)</summary>
What do you think about the nanopore assembly? is it good enough for you?
</details>

<h3 id="assembly-polishing-with-pilon">Assembly polishing with Pilon</h3>
<p>We should be able improve our assembly with the Illumina reads available and correct some of these errors.</p>
<p>This process involves two steps. We will first align the Illumina reads to our draft assembly, then supply the mapping information to Pilon, which will use this alignment information to error-correct our assembly.</p>
<p>Illumina reads have much higher per-base accuracy than Nanopore reads. We will map the Illumina reads to our draft assembly using a short-read aligner called BWA-MEM. Then we can give Pilon this alignment file to polish our draft assembly.</p>
<p><img alt="‼" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.1.0/assets/svg/203c.svg" title=":bangbang:" /> This is maybe one of the most complex steps that we are going to compute so be careful and don't get lost</p>
<p><strong>Map Illumina reads to Nanopore draft assembly</strong></p>
<p>To map the Illumina reads to the Nanopore assembly, first we need to index our assembly, for that we will be using <code>Bwa</code> and <code>Samtools</code></p>
<p>Indexing is an essential step in many bioinformatics applications, as it can greatly reduce the computational time and resources required for sequence alignment. It allows the alignment algorithm to quickly locate the query sequences in the reference genome, without having to search the entire genome for matches
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>
</span><span id="__span-11-3"><a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>assemblies/error_corrected_canu
</span><span id="__span-11-4"><a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a><span class="nb">cd</span><span class="w"> </span>assemblies/error_corrected_canu
</span><span id="__span-11-5"><a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>mkdir<span class="w"> </span>index
</span><span id="__span-11-6"><a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a>
</span><span id="__span-11-7"><a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a>bwa<span class="w"> </span>index<span class="w"> </span>-p<span class="w"> </span>index/index<span class="w"> </span>../canu/canu.contigs.fasta
</span><span id="__span-11-8"><a id="__codelineno-11-8" name="__codelineno-11-8" href="#__codelineno-11-8"></a>bwa<span class="w"> </span>mem<span class="w"> </span>index/index<span class="w"> </span>../../qc/trimmomatic/illumina_trimmed_1.fastq<span class="w">  </span>../../qc/trimmomatic/illumina_trimmed_2.fastq<span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-o<span class="w"> </span>illumina_sorted.bam
</span><span id="__span-11-9"><a id="__codelineno-11-9" name="__codelineno-11-9" href="#__codelineno-11-9"></a>samtools<span class="w"> </span>index<span class="w"> </span>-bc<span class="w"> </span>illumina_sorted.bam
</span></code></pre></div></p>
<p>We should get a BAM file as an output. This is tabular data recording information about how reads were aligned to the draft assembly.
Now we can use that the <code>illumina_sorted.bam</code>to run <code>Pilon</code>.</p>
<p><div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>pilon<span class="w"> </span>--genome<span class="w"> </span>../canu/canu.contigs.fasta<span class="w"> </span>--bam<span class="w"> </span>illumina_sorted.bam<span class="w"> </span>--outdir<span class="w"> </span>pilon<span class="w"> </span>-Xmx12G
</span></code></pre></div>
Once <code>Pilon</code> has ended we should find a unique <strong>Fasta</strong> file in the <code>pilon</code> directory</p>
<p><strong>Assembly QC</strong></p>
<p>Now that we have run Pilon, thanks to the help of the illumina reads we should get a much better assembly than before.
But we cannot call ourselves scientist only with assumptions we need facts. Therefore, let's run Busco and Quast...yes, again <img alt="😪" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.1.0/assets/svg/1f62a.svg" title=":sleepy:" /></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>assembly_qc/canu_nanopore_illumina
</span><span id="__span-13-4"><a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a><span class="nb">cd</span><span class="w"> </span>assembly_qc/canu_nanopore_illumina
</span><span id="__span-13-5"><a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a>
</span><span id="__span-13-6"><a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a>busco<span class="w"> </span>-i<span class="w"> </span>../../assemblies/error_corrected_canu/pilon/pilon.fasta<span class="w"> </span>-l<span class="w"> </span>vibrionales<span class="w"> </span>-o<span class="w"> </span>busco<span class="w"> </span>--augustus<span class="w"> </span>--mode<span class="w"> </span>genome<span class="w"> </span>--cpu<span class="w"> </span><span class="m">4</span>
</span><span id="__span-13-7"><a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a>quast<span class="w"> </span>../../assemblies/error_corrected_canu/pilon/pilon.fasta<span class="w"> </span>-r<span class="w"> </span>../../data/VP_reference_genome.fasta<span class="w"> </span>-o<span class="w"> </span>quast
</span></code></pre></div>
<p><img src="assets/Busco_pilon.png">
<img src="assets/quast_pilon.png"></p>
<details>
<summary>Question 5(click to reveal)</summary>
What do you think about the results? Do you see any improvements?
</details>

<h2 id="section-3-hybrid-assembly-unicycler">Section 3: Hybrid assembly - Unicycler</h2>
<p>In this section, we will use a purpose-built tool called <code>Unicycler</code> to perform hybrid assembly.</p>
<p><code>Unicycler</code> uses our <code>Nanopore</code> and <code>Illumina</code> read sets together as input, and returns an assembly. Once we have created the assembly, 
we will assess its quality using <code>Quast</code> and <code>Busco</code> and compare with our previous polished assembly.</p>
<p><code>Unicycler</code> performs assembly in the opposite manner to our approach. <code>Illumina</code> reads are used to create an assembly graph, then <code>Nanopore</code> reads 
are used to disentangle problems in the graph. The Nanopore reads serve to bridge Illumina contigs, and to reveal how the contigs are arranged 
sequentially in the genome. </p>
<p><strong>Run Unicycler</strong></p>
<p><div class="language-bash highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-14-2"><a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a>
</span><span id="__span-14-3"><a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a>unicycler<span class="w"> </span>-1<span class="w"> </span>qc/trimmomatic/illumina_trimmed_1.fastq<span class="w"> </span>-2<span class="w"> </span>qc/trimmomatic/illumina_trimmed_2.fastq<span class="w"> </span>-l<span class="w"> </span>qc/nanopore_trimmed/nanopore_filtered.fastq<span class="w"> </span>-o<span class="w"> </span>assemblies/unicycler<span class="w"> </span>-t<span class="w"> </span><span class="m">4</span>
</span></code></pre></div>
Once it has ended, if you go to the <code>assemblies/unicycler</code> directory, you should find the assembly file called <code>assembly.fasta</code></p>
<p><strong>Assembly QC</strong></p>
<p>As we did with the <code>Canu</code> and <code>Pilon</code> assembly, we need to perform a Quality Control (QC) to our newly assembled genome.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-15-2"><a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a>
</span><span id="__span-15-3"><a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>assembly_qc/unicycler_alone
</span><span id="__span-15-4"><a id="__codelineno-15-4" name="__codelineno-15-4" href="#__codelineno-15-4"></a><span class="nb">cd</span><span class="w"> </span>assembly_qc/unicycler_alone
</span><span id="__span-15-5"><a id="__codelineno-15-5" name="__codelineno-15-5" href="#__codelineno-15-5"></a>
</span><span id="__span-15-6"><a id="__codelineno-15-6" name="__codelineno-15-6" href="#__codelineno-15-6"></a>busco<span class="w"> </span>-i<span class="w"> </span>../../assemblies/unicycler/assembly.fasta<span class="w"> </span>-l<span class="w"> </span>vibrionales<span class="w"> </span>-o<span class="w"> </span>busco<span class="w"> </span>--augustus<span class="w"> </span>--mode<span class="w"> </span>genome
</span><span id="__span-15-7"><a id="__codelineno-15-7" name="__codelineno-15-7" href="#__codelineno-15-7"></a>quast<span class="w"> </span>../../assemblies/unicycler/assembly.fasta<span class="w"> </span>-r<span class="w"> </span>../../data/VP_reference_genome.fasta<span class="w"> </span>-o<span class="w"> </span>quast
</span></code></pre></div>
<p>these are the results that we have obtained from Unicycler:</p>
<p><img src="assets/Busco_unicycler.png">
<img src="assets/quast_unicycler.png"></p>
<details>
<summary>Question 6(click to reveal)</summary>
Which are the differnces between the pilon assembly and this one?
</details>

<details>
<summary>Question 7(click to reveal)</summary>
What do you think about the hybrid assembly? is it good enough for you?
</details>

<h2 id="section-4-nobody-expects-this-combo-unicycler-x-canu">Section 4: Nobody expects this combo - Unicycler x Canu</h2>
<p>As you have read in the section 4 title, now we are going to do a Unicycler run based on our Canu run, and yes! this is possible.
For that we are going to use the <code>Unicycler</code> option <code>--existing_long_read_assembly</code>. Remember when we mapped the illumina reads to the Canu draft assembly so<code>Pilon</code> could improve
our the assembly? Well let's say this is something similar. This time Unicycler instead of building everything without a base, it will have de Canu_improved_assembly as baseline,
and you will see that this time we will have a better result.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>
</span><span id="__span-16-3"><a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>assemblies/unicycler_canu
</span><span id="__span-16-4"><a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a><span class="nb">cd</span><span class="w"> </span>assemblies/unicycler_canu
</span><span id="__span-16-5"><a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a>
</span><span id="__span-16-6"><a id="__codelineno-16-6" name="__codelineno-16-6" href="#__codelineno-16-6"></a>unicycler<span class="w"> </span>-1<span class="w"> </span>../../qc/trimmomatic/illumina_trimmed_1.fastq<span class="w"> </span>-2<span class="w"> </span>../../qc/trimmomatic/illumina_trimmed_2.fastq<span class="w"> </span>-l<span class="w"> </span>../../qc/nanopore_trimmed/nanopore_filtered.fastq<span class="w"> </span>-o<span class="w"> </span>./<span class="w"> </span>-t<span class="w"> </span><span class="m">4</span><span class="w"> </span>--existing_long_read_assembly<span class="w"> </span>../error_corrected_canu/pilon/pilon.fasta
</span></code></pre></div>
<p>I can't believe it! We are at the end, just compute the QC of that assembly and let's go back home <img alt="😖" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.1.0/assets/svg/1f616.svg" title=":confounded:" /></p>
<p><div class="language-bash highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="c1">#go to the main directory hybrid_assembly</span>
</span><span id="__span-17-2"><a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a>
</span><span id="__span-17-3"><a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>assembly_qc/unicycler_canu
</span><span id="__span-17-4"><a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a><span class="nb">cd</span><span class="w"> </span>assembly_qc/unicycler_canu
</span><span id="__span-17-5"><a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a>
</span><span id="__span-17-6"><a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a>busco<span class="w"> </span>-i<span class="w"> </span>../../assemblies/unicycler_canu/assembly.fasta<span class="w"> </span>-l<span class="w"> </span>vibrionales<span class="w"> </span>-o<span class="w"> </span>busco<span class="w"> </span>--augustus<span class="w"> </span>--mode<span class="w"> </span>genome<span class="w"> </span>--cpu<span class="w"> </span><span class="m">4</span>
</span><span id="__span-17-7"><a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a>quast<span class="w"> </span>../../assemblies/unicycler_canu/assembly.fasta<span class="w"> </span>-r<span class="w"> </span>../../data/VP_reference_genome.fasta<span class="w"> </span>-o<span class="w"> </span>quast
</span></code></pre></div>
<img src="assets/Busco_unicyclerXcanu.png">
<img src="assets/quast_unicyclerXcanu.png"></p>







  
  






                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 EAOMICS based on the Hybrid genome assembly page of Melbourne Bioinformatics
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": ".", "features": ["navigation.tabs", "navigation.sections", "toc.integrate", "navigation.top", "search.suggest", "search.highlight", "content.tabs.link", "content.code.annotation", "content.code.copy"], "search": "assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="assets/javascripts/bundle.83f73b43.min.js"></script>
      
    
  </body>
</html>